subprojects/language-native/src/main/java/org/gradle/language/swift/tasks/SwiftCompile.java

```diff
@@ -78,4 +80,18 @@ public void setModuleName(String moduleName) {
     public void setModuleName(Provider<String> moduleName) {
         this.moduleName.set(moduleName);
     }
+
+    @Override
+    public void compile(IncrementalTaskInputs inputs) {
+        SimpleStaleClassCleaner cleaner = new SimpleStaleClassCleaner(getOutputs());
+        cleaner.setDestinationDir(getObjectFileDir());
+        cleaner.execute();
+
+        if (getSource().isEmpty()) {
```

Comment:
I think this might be unnecessary if `getSource` is annotated with `@SkipWhenEmpty`. I think we delete the outputs when all the sources are removed.

Reply:
I remove that part and didn't add `@SkipWhenEmpty` as removing all source will be tested in another PR.